src/daily_output.ts

```diff
@@ -226,40 +226,40 @@ export class DailyOutput {
 		const node = edge.node;
 		const txId = node.id;
 		const ownerAddress = node.owner.address;
-		const tip = +node.quantity.winston;
 		const tags = node.tags;
 		const dataSize = +node.data.size;
-		const entityTypeTag = tags.find((tag) => tag.name === 'Entity-Type')?.value;
-		const bundledIn = node.bundledIn?.id;
-		const bundleVersion = tags.find((tag) => tag.name === 'Bundle-Version')?.value;
-		const bundleTipType = tags.find((tag) => tag.name === 'Tip-Type')?.value;
-		const isMetadataTransaction = !!entityTypeTag && !bundleVersion;
-		const isBundleTransaction = !!bundleVersion;
-
-		const previousTimestamp = this.latestTimestamp * 1000;
-		const previousBlockHeight = this.previousData.blockHeight;
-		const previousDate = new Date(previousTimestamp);
 
 		const height = edge.node.block.height;
-
+		const previousBlockHeight = this.previousData.blockHeight;
 		if (previousBlockHeight >= height) {
 			throw new Error('That block was already processed!');
 		}
 
 		const queryTimestamp = edge.node.block.timestamp * 1000;
 		const queryDate = new Date(queryTimestamp);
-
+		const previousTimestamp = this.latestTimestamp * 1000;
+		const previousDate = new Date(previousTimestamp);
 		if (this.isNewESTDate(previousDate, queryDate)) {
 			this.resetDay();
 		}
-
 		if (this.isNewESTWeek(previousDate, queryDate)) {
 			this.resetWeek();
 		}
 
+		const boostValue = +(tags.find((tag) => tag.name === 'Boost')?.value || '1');
+		const tip = +node.quantity.winston;
+		const fee = +node.fee.winston;
+		const unboostedFee = fee / boostValue;
+		const tipPercentage = unboostedFee / tip;
```

Comment:
Not sure if this is calculating out correctly. e.g: reward 20000 tip of 5000 -- `20000 / 5000 = 4%`

Reply:
I think its just `tip / unboostedFee` here

Reply:
We should absolutely extract this tip validation out and unit test it. We want to know exactly which inputs would succeed and which would fail

Reply:
I wasn't calculating the percentage, but the fee multiple. Fixed now. 
[f125f1f](https://github.com/ardriveapp/inferno-backend/pull/15/commits/f125f1fc8466593ae5a4cadb89d7044510a4426f)

Reply:
This is really good, but what I was hoping for some tests around `isTipPercentageValid`. I'd like to know what tipPercentage values would be considered invalid and which would be valid

Reply:
Something like this: 

```ts

export function isTipPercentageValid(tip: number, tipPercentage: number): boolean {
	if (!tip) {
		return false;
	}
	// we are using EPSILON here to have a minimum range of error acceptance
	const tipWithEpsilon = tipPercentage + Number.EPSILON;
	return tipWithEpsilon >= 15;
}
```

Reply:
But I think we get most of what I was looking for with your changes.